bukkit/src/main/java/net/william278/husksync/BukkitHuskSync.java

```diff
@@ -120,7 +122,17 @@ public void onEnable() {
             }
 
             // Prepare database connection
-            this.database = new MySqlDatabase(settings, resourceReader, logger, dataAdapter, eventCannon);
+            DatabaseType databaseType = DatabaseType.valueOf(settings.getStringValue(Settings.ConfigOption.DATABASE_TYPE));
```

Comment:
I'd prefer a static method in the `DatabaseType` enum that returns an `Optional<>` databaseType as a string. You could then run an IfPresentOrElse and throw the `HuskSyncInitializationException` in the `else` lambda expression.

Example, in `DataSaveCause`:
https://github.com/WiIIiam278/HuskSync/blob/460cb54a7dd1ad4e8b28b0d2e90f46a66eda85a0/common/src/main/java/net/william278/husksync/data/DataSaveCause.java#L87-L95